src/azure-cli/azure/cli/command_modules/acr/_params.py

```diff
@@ -434,6 +434,10 @@ def load_arguments(self, _):  # pylint: disable=too-many-statements
         c.argument('ignore_errors', options_list=['--ignore-errors'], help='Provide all health checks, even if errors are found', action='store_true', required=False)
         c.argument('vnet', options_list=['--vnet'],
                    help="Virtual network ID so to run this command inside a VNET to verify the DNS routing to private endpoints", required=False)
+        c.argument('registry_name', options_list=['--name', '-n'], help="The name of the registry.")
+        c.argument('repository_name', options_list=['--repository', '-r'], help="The target repository namespace such as 'ubuntu'.")
+        c.argument('image', options_list=['--image','-t'],
```

Comment:
We already have a definition for this variable. Is there a reason for this to be custom?
Same goes for the other added parameters

Reply:
@rosanch Looks like majority of the commands actually use 'registry_name', options_list=['--registry', '-r'] instead of ['--name', '-n']. This '--name' was being used prior to my changes, and is also being used on line 82:
![image](https://github.com/Azure/azure-cli/assets/82487617/64383d28-afb7-41d0-88cf-4a6df47c7aa1)

I can update the check-health command to use ''--registry, -r " if that is the most current version.

Also, the already existing variable, 'image_by_tag_or_digest_type', uses [--image, 't'] but it is defined as "The name of the image. May include a tag in the format 'name:tag' or digest in the format 'name@digest'." It combines the repository name with tag or digest.

The reason I defined --image differently is because with this PR, the check-health command asks user for the repository name, then the image or tag separately and '--image, -t' in this case means either the digest itself ("sha256:123") or the tag ("latest").  I asked for repo name separately as it is used in the APIs for 'get blob' and 'get manifest'.

Alternatively, I could use '--digest' and '--tag' to make it more precise, or refactor code to use '--image' as previously defined and parse out the repo name in the code.
There were concerns that users would not always quickly recall the image digest and that '--image' might be a better blanket term. Open to suggestions and adding @estebanreyl and @AntonioVT  to this comment as they had some input into this.

Reply:
Then I don't think it should be name image. Let's keep consistency normally we request for the repo or the image not both and we add validation for it. If you want to use a different format, we can't call it the same. For example, if it is only tag or digest, we can call it reference as shown in our API documentation. 
https://learn.microsoft.com/en-us/rest/api/containerregistry/manifests/get?view=rest-containerregistry-2019-08-15&tabs=HTTP
If you don't want to introduce a new pattern to the CLI and get in touch with the PM's to validate it. Then please use the existing image/repo pattern we use in all other commands

Reply:
@rosanch - yes, so then the command would be rewritten as  'acr check-health --registry myregistry --image hello-world@sha256:123 (or -t hello-world:latest). I will check with the PMs if we can change to '--registry, -r' instead of '--name, -n' or if they'd like to keep it to '--name, -n'. @terencet-dev 

question though - in the new artifact streaming command, we are using '--name' for registry. I'm guessing this is because the command also uses '--repository' and it would be potentially confusing?

Reply:
Consulted with Terence and we will keep it to '--name, -n' to maintain consistency with the heavily used commands like 'acr create', 'acr login', 'acr import' etc. that use '--name' for the registry name.